### PR TITLE
Added MCP3008

### PR DESCRIPTION
**Pull request by SenexCrenshaw**
## Description:
Added doc for mcp3008

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#1057

## Checklist:

  - [X ] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [ X] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/mcp3008rst [hunk 1]
```diff
@@ -0,0 +1,70 @@
+MCP3008 I/O Expander
```
Comment (glmnet) [inline components/mcp3008rst]: file name is missing the dot in .rst

File: components/mcp3008rst [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        resistor: 10kOhm
+        name: Resistance Sensor
+      - platform: ntc
+        id: freezer_temp
+        sensor: resistance_sensor
+        calibration:
+          b_constant: 3950
+          reference_temperature: 25°C
+          reference_resistance: 10kOhm
+        name: NTC Temperature
```
Comment (glmnet) [inline components/mcp3008rst:56]: why are you adding this? is it related to the port expander?

File: index.rst [hunk 3]
```diff
@@ -329,6 +329,7 @@ Misc Components
 
     PCF8574 I/O Expander, components/pcf8574, pcf8574.jpg
     MCP230XX I/O Expander, components/mcp230xx, mcp230xx.svg
+    MCP3008 8-Channel 10-Bit A/D Converter, components/mcp3008, mcp3008.png
```
Comment (glmnet) [inline index.rst:332]: ideally we use a very small (bit size) image on the index and if you want a bigger one on the component page. Please use image optimization sw as tinyjpg.com 
More info is available in the contrib guide on esphome docs.

File: components/mcp3008rst [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+Configuration variables:
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+- **id** (**Required**, :ref:`config-id`): The id to use for this MCP3008 component.
+- **cs_pin** (*Optional*, int): The SPI cable select pin to use
```
Comment (glmnet) [inline components/mcp3008rst]: There are more configuration parameters available which should be listed

File: components/mcp3008rst [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        update_interval: 1s
+        mcp3008: 'mcp3008_hub'
+        name: "FridgeTmep"
+        number: 1
+      - platform: resistance
+        id: resistance_sensor
+        sensor: freezer_temp_source
+        configuration: DOWNSTREAM
+        resistor: 10kOhm
+        name: Resistance Sensor
```
Comment (glmnet) [inline components/mcp3008rst]: hide name here, so resistance sensor does not go to HA

File: components/mcp3008rst [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+      - id: 'MCP3008_hub'
+        cs_pin: D8
+
+    # Individual outputs
+    sensor:
+      - platform: mcp3008
+        update_interval: 1s
+        mcp3008: 'mcp3008_hub'
+        id: freezer_temp_source
+        name: "Freezer Temp"
```
Comment (glmnet) [inline components/mcp3008rst]: hide name here, so Freezer Temp does not go to HA in a weird raw value

File: components/mcp3008rst [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+        number: 0
+      - platform: mcp3008
+        update_interval: 1s
+        mcp3008: 'mcp3008_hub'
+        name: "FridgeTmep"
```
Comment (glmnet) [inline components/mcp3008rst]: Check spelling, also, is this a good example? what will the user see here on the UI?

File: components/mcp3008rst (hunk 2 above)
Comment (glmnet) [inline components/mcp3008rst]: I'd make sense to name this one "Fridge temp"

File: components/mcp3008rst [hunk 8]
```diff
@@ -0,0 +1,70 @@
+MCP3008 I/O Expander
+=====================
+
+.. seo::
+    :description: Instructions for setting up MCP3008 10 Bit Analog to Digital Converter in ESPHome.
+    :image: mcp3008.png
+
+The Microchip Technology Inc. MCP3008
+devices are successive approximation 10-bit Analogto-Digital (A/D) converters with on-board sample and
+hold circuitry
```
Comment (glmnet) [inline components/mcp3008rst]: You should say somewhere what the value read on the pins are? Which is the voltage reference for the sampling? Etc.

**Review by glmnet — CHANGES_REQUESTED** [review]

File: components/mcp3008rst (hunk 4 above)
Comment (SenexCrenshaw) [reply components/mcp3008rst]: I just need the id and cs pin. What else should I add? I posted the same question in the component pr as well.

File: index.rst (hunk 3 above)
Comment (SenexCrenshaw) [reply index.rst:332]: Resized in .5

File: components/mcp3008rst (hunk 1 above)
Comment (SenexCrenshaw) [reply components/mcp3008rst]: doh!

File: components/mcp3008rst (hunk 5 above)
Comment (SenexCrenshaw) [reply components/mcp3008rst]: talk about confusion. I guess i was excited to send this PR in. I typically dont contribute but am trying to be a better open source developer. Also building a home thermostat and need some components and love esphome so... some PRs from me. 

 # Example configuration entry
    MCP3008:
      - id: 'MCP3008_hub'
        cs_pin: D8

    # Example config of sensors. 
    # This is a NTCB3950 10K thermocoupler attached to pin 0
    # of the MCP3008 with a 10K resistor as a voltage divider. 
    # See `resistance` and `ntc` platorms for other options
    sensor:
      - platform: mcp3008             # Attached to pin 0 of the MCP3008. The result will be a reading from 0-1023
        update_interval: 1s
        mcp3008: 'mcp3008_hub'
        id: freezer_temp_source       
        number: 0                     # pin number
      - platform: resistance
        id: freezer_resistance_sensor 
        sensor: freezer_temp_source
        configuration: DOWNSTREAM
        resistor: 10kOhm
        name: Resistance Sensor
      - platform: ntc
        id: freezer_temp
        sensor: freezer_resistance_sensor
        calibration:
          b_constant: 3950
          reference_temperature: 25°C
          reference_resistance: 10kOhm
        name: Freezer Temperature

**Pull request merged**